mirrord/config/derive/src/config/field.rs

```diff
@@ -173,10 +170,23 @@ impl ConfigField {
                 None => quote! { None }
             };
 
-            quote! { .ok_or(crate::config::ConfigError::ValueNotProvided(stringify!(#parent), stringify!(#ident), #env_override))? }
+            // unwrap to default if exists
+            if let Some(default) = flags.default.as_ref() {
+                quote! {#default}
+            } else {
+                quote! { .ok_or(crate::config::ConfigError::ValueNotProvided(stringify!(#parent), stringify!(#ident), #env_override))? }
+            }
         });
 
-        quote! { #ident: (#(#impls),*) #(#layers)* .source_value()#unwrapper }
+        let impls = impls
+            .into_iter()
+            .reduce(|acc, impl_| quote! { #acc.or(#impl_) });
+
+        let layers = layers
+            .into_iter()
+            .reduce(|acc, layer| quote! { #acc.or(#layer) });
+
+        quote! { #ident: #impls #layers .source_value().transpose()?#unwrapper }
```

Comment:
layers already contains the call to `.layer(Fn())` will not work with `.or(Fn())`

```suggestion
        quote! { #ident: #impls (#(#impls),*) .source_value().transpose()?#unwrapper }
```

Reply:
double impls? wat?

Reply:
ops it should be layers